pkg/idp/mock_test.go

```diff
@@ -15,3 +23,38 @@ func toJSON(data any) string {
 	bytes, _ := json.Marshal(data)
 	return string(bytes)
 }
+
+type componentMocks struct {
+	mockCtrl          *gomock.Controller
+	keycloakIdpClient *mock.KeycloakIdpClient
+	tokenProvider     *mock.OidcTokenProvider
+	hrdTool           *mock.ComponentTool
+	component         *mock.Component
+	logger            *mock.Logger
+}
+
+func createMocks(t *testing.T) *componentMocks {
+	mockCtrl := gomock.NewController(t)
+	return &componentMocks{
+		mockCtrl:          mockCtrl,
+		keycloakIdpClient: mock.NewKeycloakIdpClient(mockCtrl),
+		tokenProvider:     mock.NewOidcTokenProvider(mockCtrl),
+		hrdTool:           mock.NewComponentTool(mockCtrl),
+		component:         mock.NewComponent(mockCtrl),
+		logger:            mock.NewLogger(mockCtrl),
+	}
+}
+
+func (m *componentMocks) finish() {
+	m.mockCtrl.Finish()
+}
+
+func (m *componentMocks) createComponent() *component {
+	return NewComponent(m.keycloakIdpClient, m.tokenProvider, m.hrdTool, m.logger).(*component)
+}
+
+func (m *componentMocks) newEndpoints() Endpoints {
+	return NewEndpoints(m.component, func(e cs.Endpoint, name string) endpoint.Endpoint {
+		return endpoint.Endpoint(e)
```

Comment:
Is it something we should do now? I think those functions used to be in `component_test.go`